src/common/types/metadata.js

```diff
@@ -114,7 +114,29 @@ type MetadataSuggestion = {
     suggestions: { [key: string]: string | number | string[] },
 };
 
+type MetadataInstanceTemplateField = {
+    description?: string,
+    displayName: string,
+    hidden?: boolean,
+    id: string,
+    key: string, // V2
+    options?: Array<MetadataTemplateFieldOption>, // V3
+    type: MetadataFieldType,
+    value: string,
+};
+
+type MetadataTemplateInstance = {
+    displayName?: string,
+    hidden?: boolean,
+    id: string,
+    metadataFields: MetadataInstanceTemplateField[],
```

Comment:
let's keep the naming consistent
```suggestion
    fields: MetadataInstanceTemplateField[],
```